packages/opentelemetry-exporter-collector/src/CollectorExporter.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { Attributes, Logger } from '@opentelemetry/types';
+
+import * as collectorTypes from './types';
+import { toCollectorSpan } from './transform';
+
+import { onInit, onShutdown, sendSpans } from './platform/index';
+
+/**
+ * Collector Exporter Config
+ */
+export interface CollectorExporterConfig {
+  hostName?: string;
+  logger?: Logger;
+  serviceName?: string;
+  spanAttributes?: Attributes;
+  url?: string;
+}
+
+const defaultServiceName = 'collector-exporter';
+const defaultCollectorUrl = 'http://localhost:55678/v1/trace';
+
+/**
+ * Collector Exporter
+ */
+export class CollectorExporter implements SpanExporter {
+  readonly serviceName: string;
+  readonly url: string;
+  readonly logger: Logger;
+  readonly hostName: string | undefined;
+  private _isShutdown: boolean = false;
+
+  /**
+   * @param config
+   */
+  constructor(config: CollectorExporterConfig = {}) {
+    this.serviceName = config.serviceName || defaultServiceName;
+    this.url = config.url || defaultCollectorUrl;
+    if (typeof config.hostName === 'string') {
+      this.hostName = config.hostName;
+    }
+
+    this.logger = config.logger || new NoopLogger();
+
+    this.shutdown = this.shutdown.bind(this);
```

Comment:
You don't need this line if you make `shutdown` an arrow function property of `CollectorExporter`:

```typescript
export class CollectorExporter implements SpanExporter {
  ...
  shutdown = (): void => {
    ...
  };
  ...
}
```

Just a matter of style so if you don't like it then leave it as is. I like to do it this way because then I don't have to remember to call `.bind(this)`